src/uct/ugni/rdma/ugni_rdma_iface.c

```diff
@@ -161,54 +161,59 @@ static UCS_CLASS_CLEANUP_FUNC(uct_ugni_rdma_iface_t)
 static UCS_CLASS_DEFINE_DELETE_FUNC(uct_ugni_rdma_iface_t, uct_iface_t);
 
 static uct_iface_ops_t uct_ugni_aries_rdma_iface_ops = {
-    .iface_query         = uct_ugni_rdma_iface_query,
-    .iface_flush         = uct_ugni_iface_flush,
-    .iface_close         = UCS_CLASS_DELETE_FUNC_NAME(uct_ugni_rdma_iface_t),
-    .iface_get_address   = uct_ugni_iface_get_address,
-    .iface_get_device_address = uct_ugni_iface_get_dev_address,
-    .iface_is_reachable  = uct_ugni_iface_is_reachable,
-    .ep_create_connected = UCS_CLASS_NEW_FUNC_NAME(uct_ugni_ep_t),
-    .ep_destroy          = UCS_CLASS_DELETE_FUNC_NAME(uct_ugni_ep_t),
     .ep_put_short        = uct_ugni_ep_put_short,
     .ep_put_bcopy        = uct_ugni_ep_put_bcopy,
     .ep_put_zcopy        = uct_ugni_ep_put_zcopy,
+    .ep_get_bcopy        = uct_ugni_ep_get_bcopy,
+    .ep_get_zcopy        = uct_ugni_ep_get_zcopy,
     .ep_am_short         = uct_ugni_ep_am_short,
     .ep_atomic_add64     = uct_ugni_ep_atomic_add64,
     .ep_atomic_fadd64    = uct_ugni_ep_atomic_fadd64,
     .ep_atomic_cswap64   = uct_ugni_ep_atomic_cswap64,
-    .ep_get_bcopy        = uct_ugni_ep_get_bcopy,
-    .ep_get_zcopy        = uct_ugni_ep_get_zcopy,
-    .ep_pending_add      = uct_ugni_ep_pending_add,
-    .ep_pending_purge    = uct_ugni_ep_pending_purge,
     .ep_atomic_swap64    = uct_ugni_ep_atomic_swap64,
     .ep_atomic_add32     = uct_ugni_ep_atomic_add32,
     .ep_atomic_fadd32    = uct_ugni_ep_atomic_fadd32,
     .ep_atomic_cswap32   = uct_ugni_ep_atomic_cswap32,
     .ep_atomic_swap32    = uct_ugni_ep_atomic_swap32,
+    .ep_pending_add      = uct_ugni_ep_pending_add,
+    .ep_pending_purge    = uct_ugni_ep_pending_purge,
     .ep_flush            = uct_ugni_ep_flush,
-};
-
-static uct_iface_ops_t uct_ugni_gemini_rdma_iface_ops = {
-    .iface_query         = uct_ugni_rdma_iface_query,
+    .ep_fence            = uct_base_ep_fence,
+    .ep_create_connected = UCS_CLASS_NEW_FUNC_NAME(uct_ugni_ep_t),
+    .ep_destroy          = UCS_CLASS_DELETE_FUNC_NAME(uct_ugni_ep_t),
     .iface_flush         = uct_ugni_iface_flush,
+    .iface_fence         = uct_base_iface_fence,
+    .iface_progress_enable = ucs_empty_function,
+    .iface_progress_disable = ucs_empty_function,
```

Comment:
These two lines are the problem. It looks like there is a definition in uct.h for uct_iface_progress_enable() but uct_iface_ops_t does not have a .iface_progress_enable or a .iface_progress_disable. Removing these fixes everything and the tests pass.